gpt_index/readers/slack.py

```diff
@@ -64,6 +65,8 @@ def _read_message(self, channel_id: str, message_ts: str) -> str:
                 next_cursor = result["response_metadata"]["next_cursor"]
             except SlackApiError as e:
```

Comment:
is there a way to specifically catch a rate limit error? SlackApiError seems to be a catch all

that way we sleep if it's a rate limit error, but otherwise throw an error

Reply:
yeah. Will check in API docs and update.

Reply:
```python
if e.response["error"] == "rate_limited":
            # wait for the rate limit to reset
            time.sleep(e.response["retry_after"])
        else:
            # not a rate limit error, raise the error
            raise e
```

Reply:
Thanks @simonManydata. Will update the PR.